test_scripts/RC/OnRCStatus/commonOnRCStatus.lua

```diff
@@ -0,0 +1,250 @@
+---------------------------------------------------------------------------------------------------
+-- OnRCStatus common module
+---------------------------------------------------------------------------------------------------
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+config.defaultProtocolVersion = 2
+config.ValidateSchema = false
+config.application1.registerAppInterfaceParams.appHMIType = { "REMOTE_CONTROL" }
+config.application2.registerAppInterfaceParams.appHMIType = { "REMOTE_CONTROL" }
+
+--[[ Required Shared libraries ]]
+local commonRC = require('test_scripts/RC/commonRC')
+local test = require("user_modules/dummy_connecttest")
+
+--[[ Local Variables ]]
+local commonOnRCStatus = {}
+commonOnRCStatus.modules = {
+  "CLIMATE",
+  "RADIO",
+  "AUDIO",
+  "LIGHT",
+  "HMI_SETTINGS",
+  "SEAT"
+}
+
+function commonOnRCStatus.getRCAppConfig()
+  local struct = commonRC.getRCAppConfig()
+  -- struct.moduleType = { "RADIO", "CLIMATE", "AUDIO", "LIGHT", "HMI_SETTINGS", "SEAT" }
+  struct.moduleType = { "RADIO", "CLIMATE" }
+  return struct
+end
+
+function commonOnRCStatus.getMobileSession(pAppId)
+  return commonRC.getMobileSession(test, pAppId)
+end
+
+function commonOnRCStatus.preconditions()
+  commonRC.preconditions()
+end
+
+function commonOnRCStatus.start()
+  commonRC.start(nil, test)
+end
+
+local function PTUfunc(tbl)
+  local appId = config.application1.registerAppInterfaceParams.appID
+  tbl.policy_table.app_policies[appId] = commonOnRCStatus.getRCAppConfig()
+end
+
+function commonOnRCStatus.RegisterRCapplication(pModuleStatus, ptu_update_func, pAppId)
+  if not pAppId then pAppId = 1 end
+  if not pModuleStatus then pModuleStatus = { freeModules = commonOnRCStatus.modules, allocatedModules = { } } end
+  if not ptu_update_func then
+    ptu_update_func = PTUfunc
+  end
+  commonRC.rai_ptu_n(pAppId, ptu_update_func, test)
+  local mobSession = commonOnRCStatus.getMobileSession(pAppId)
+  mobSession:ExpectNotification("OnRCStatus",pModuleStatus)
+  pModuleStatus.appID = commonRC.getHMIAppId()
+  EXPECT_HMINOTIFICATION("RC.OnRCStatus", pModuleStatus )
+end
+
+function commonOnRCStatus.rai_ptu_n(ptu_update_func, pAppId)
+  if not pAppId then pAppId = 1 end
+  if not ptu_update_func then
+    ptu_update_func = PTUfunc
+  end
+  commonRC.rai_ptu_n(pAppId, ptu_update_func, test)
+end
+
+function commonOnRCStatus.rai_n_rc_app(id)
+  commonRC.rai_n(id, test)
+end
+
+function commonOnRCStatus.SubscribeToModule(pModule, pAppId)
+  if not pAppId then pAppId = 1 end
+  commonRC.subscribeToModule(pModule, pAppId, test)
+  local mobSession = commonOnRCStatus.getMobileSession(pAppId)
+  mobSession:ExpectNotification("OnRCStatus")
+  :Times(0)
+  EXPECT_HMINOTIFICATION("RC.OnRCStatus")
+  :Times(0)
+end
+
+function commonOnRCStatus.SubscribeToModuleWithoutNot(pModule, pAppId)
```

Comment:
@GetmanetsIrina Function is not used in scripts

Reply:
Locally changed.